### PR TITLE
[spinel] allow custom config file for lib spinel

### PR DESCRIPTION
**Pull request by Irving-cl**
This PR allows user projects to override the ot lib configs by using
a customized lib config file, by setting `OT_LIB_CONFIG`.

### CODE REVIEW

**Comment by Irving-cl** [comment]
@gytxxsy This PR allows to specify the spinel config file.

File: src/lib/spinel/CMakeLists.txt [hunk 1]
```diff
@@ -26,6 +26,14 @@
 #  POSSIBILITY OF SUCH DAMAGE.
 #
 
+add_library(ot-lib-spinel-config INTERFACE)
+
+set(OT_LIB_SPINEL_CONFIG "" CACHE STRING "Lib spinel project-specific config header file")
+if (OT_LIB_SPINEL_CONFIG)
```
Comment (zhanglongxia) [inline src/lib/spinel/CMakeLists.txt:32]: Here only specify the custom config file for the Spinel module, so we need to add extra custom config file configurations for other lib modules in the future. I think we can unify the config file for all lib modules.

Comment (Irving-cl) [reply src/lib/spinel/CMakeLists.txt:32]: Make sense. Any suggestion on the naming for this macro? What about using `OT_LIB_CONFIG`?

Comment (Irving-cl) [reply src/lib/spinel/CMakeLists.txt]: I updated to make the config for the entire lib. Currently I name it as `OT_LIB_CONFIG`.

File: src/lib/CMakeLists.txt [hunk 2]
```diff
@@ -26,6 +26,14 @@
 #  POSSIBILITY OF SUCH DAMAGE.
 #
 
+add_library(ot-lib-config INTERFACE)
+
+set(OT_LIB_CONFIG "" CACHE STRING "Lib spinel project-specific config header file")
```
Comment (zhanglongxia) [inline src/lib/CMakeLists.txt:31]: nit: `Lib project-specific config header file`

**Review by zhanglongxia — APPROVED** [review]
LGTM

File: src/lib/CMakeLists.txt (hunk 2 above)
Comment (zhanglongxia) [reply src/lib/CMakeLists.txt:31]: Please also update the commit message.

**Review by abtink — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**